website/src/content/api-reference/commercetools-frontend-application-shell-connectors.mdx

```diff
@@ -0,0 +1,314 @@
+
+Let's say you have something like this `custom-application-config.mjs` file in your project:
+
+```json
+{
+  "name": "My Custom Application",
+  "entryPointUriPath": "my-custom-application",
+  "cloudIdentifier": "gcp-eu",
+  "additionalEnv": {
+    "trackingSentry": "https://000@sentry.io/000",
+  },
+  ... // rest of your configuration
+}
+```
+
+Your `context.environment` object will look like this:
+
+```json
+{
+  "environment": {
+    "applicationId": "ckw3vt1hv034901uzio4bkzc3:my-custom-application",
+    "applicationName": "My Custom Application",
+    "entryPointUriPath": "my-custom-application",
+    "trackingSentry": "https://000@sentry.io/000",
+  }
+}
+```
+
+<Info>
+
+More information about accessing your custom configuration can found [here](/api-reference/application-config#runtime-application-environment)
+
+</Info>
+
+## Usage
+
+### useApplicationContext
+
+This is a custom hook which allows to access the `ApplicationContext` from any component (or other hooks).
+
+```jsx
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+
+const DisplayLocale = () => {
+  const { locale } = useApplicationContext((context) => ({
+    locale: applicationContext.dataLocale
+  }));
+  render() {
+    return (
+      <div>
+        <h1>Current locale: {locale}</h1>
+      </div>
+    );
+  }
+}
+
+export default DisplayLocale;
+```
+
+### withApplicationContext
+
+This is a High Order Component which allows to access the `ApplicationContext` from any component.
+It is meant to be used if you have a class based component where you need to read context data.
+
+```jsx
+import { withApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+
+class DisplayLocale extends React.Component {
+  render() {
+    return (
+      <div>
+        <h1>Current locale: {this.props.locale}</h1>
+      </div>
+    );
+  }
+}
+
+export default withApplicationContext((applicationContext) => ({
+  locale: applicationContext.dataLocale
+}))(DisplayLocale);
+```
+
+# Project image settings
+
+When using your images for your projects you are expected to upload them to our platform. However, you can also choose to host your images in another service.<br/>
+If this is the case, we don't show them by default to prevent for huge images to be loaded in the browser that could impact Merchat Center performance.
+
+To enable the rendering of those images you need to set up a project-level configuration where you specify rewrite rules for different images sizes.<br/>
+You can find this configuration in the settings section of your projects in the Merchant Center (**Settings > Project Settings > Miscellaneous**).
+
+## Usage
+
+### ProjectExtensionProviderForImageRegex
+
+If you need to access the rewrite rules for your project images, you need to use this provider in your component's tree.<br/>
+The provider will load the configuration and make it available using the components listed below.
```

Comment:
```suggestion
This is the React context provider that loads the image regex configuration and expose it in the React context so that children can access the configuration.

<Info>

This component is required to be defined in a parent component where children of this component need to access the configuration.

</Info>
```